### PR TITLE
Add fin-python-library page to docs

### LINKED ISSUE #126: Add documentation SOARCA-fin python library
Add documentation from the SOARCA-fin python library into the documentation on this project.

### PR DESCRIPTION
**Pull request by xncz8h**
closed #126

### CODE REVIEW

File: docs/content/en/docs/soarca-extensions/fin-python-library.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+class Parser
+class Executor
+
+ISoarcaFin <|.. SoarcaFin
+IMQTTClient <|.. MQTTClient
+IParser <|.. Parser
+IExecutor <|.. Executor
+
+IMQTTClient <- SoarcaFin
+IExecutor <- MQTTClient
```
Comment (MaartendeKruijf) [inline docs/content/en/docs/soarca-extensions/fin-python-library.md:64]:
```suggestion
MQTTClient -> IExecutor
```

File: docs/content/en/docs/soarca-extensions/fin-python-library.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+To use the SOARCA Fin library use the following command to install the library using pip:
+```bash
+pip install soarca-fin-library
+```
+
+### Example
+An example on how to use the library is given below. 
+For more examples and the source code, we will refer to the Github page of the [project](https://github.com/COSSAS/SOARCA-FIN-python-library).
+```python
+def capability_pong_callback(command: Command) -> ResultStructure:
```
Comment (MaartendeKruijf) [inline docs/content/en/docs/soarca-extensions/fin-python-library.md:23]: Maybe define the variables like finId, capabilityId and use them at the correct places?

File: docs/content/en/docs/soarca-extensions/fin-python-library.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+title:  Fin Python Library
+description: >
+    Documentation of the Python Fin library
+categories: [extensions, architecture]
+tags: [fin, python]
+weight: 2
+date: 2024-04-10
+---
+
+For documentation about the fin protocol we refer the reader to de documention page of [SOARCA](https://cossas.github.io/SOARCA/docs/soarca-extensions/fin-protocol/)
```
Comment (RabbITCybErSeC) [inline docs/content/en/docs/soarca-extensions/fin-python-library.md:11]: @xncz8h please fix typo: the reader to `the` documentation page. Plus change link name SOARCA to `Fin Protocol`

File: docs/content/en/docs/soarca-extensions/fin-python-library.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+
+## Quick Start
+To use the SOARCA Fin library use the following command to install the library using pip:
```
Comment (RabbITCybErSeC) [inline docs/content/en/docs/soarca-extensions/fin-python-library.md:14]: Please change to: To include the SOARCA Fin library, you can use the following command to install it via pip:

File: docs/content/en/docs/soarca-extensions/fin-python-library.md [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+```bash
+pip install soarca-fin-library
+```
+
+### Example
+An example on how to use the library is given below. 
+For more examples and the source code, we will refer to the Github page of the [project](https://github.com/COSSAS/SOARCA-FIN-python-library).
+```python
+
```
Comment (RabbITCybErSeC) [inline docs/content/en/docs/soarca-extensions/fin-python-library.md:23]: remove empty line

File: docs/content/en/docs/soarca-extensions/fin-python-library.md [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+## Quick Start
+To use the SOARCA Fin library use the following command to install the library using pip:
+```bash
+pip install soarca-fin-library
+```
+
+### Example
+An example on how to use the library is given below. 
+For more examples and the source code, we will refer to the Github page of the [project](https://github.com/COSSAS/SOARCA-FIN-python-library).
```
Comment (RabbITCybErSeC) [inline docs/content/en/docs/soarca-extensions/fin-python-library.md:21]: Change project to `SOARCA-Fin-python-library` in link

File: docs/content/en/docs/soarca-extensions/fin-python-library.md [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+    USERNAME = os.getenv("MQTT_USERNAME", "soarca")
+    PASSWD = os.getenv("MQTT_PASSWD", "password")
+
+    main(MQTT_BROKER, MQTT_PORT, USERNAME, PASSWD)
+```
+
+## Architecture
+### Application Layout
+The main object of the application is the `SoarcaFin` object, which is responsible for configuring and creating and controlling the capabilities.
+The SoarcaFin creates `MQTTClient`s for each capability registered, plus one for registering, unregistering and controlling the fin.
```
Comment (RabbITCybErSeC) [inline docs/content/en/docs/soarca-extensions/fin-python-library.md:97]: The SoarcaFin creates `MQTTClient`s for each capability registered, plus one for registering, unregistering and controlling the fin `itself`.

File: docs/content/en/docs/soarca-extensions/fin-python-library.md [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+`MQTTClient`s each have their own connection to the MQTT Broker and own `Parser` and `Executor` objects.
+The `Parser` object parsers the raw MQTT messages and tries to convert them to one of the objects in `src/models`.
+The `Executor` runs in their own thread and handles the actual execution of the messages.
+The `Executor` polls a thread-safe queue for new messages and performs IO operations, such as sending messages to the MQTT broker and calling capability callbacks.
+
+### Setup SOARCA Capabilities
+To register a fin to SOARCA, first create a `SoarcaFin` object and pass the `fin_id` in the constructor.
```
Comment (RabbITCybErSeC) [inline docs/content/en/docs/soarca-extensions/fin-python-library.md:104]: Are the specific requirements to the `fin_id`? should this be in a certain format? please explain

File: docs/content/en/docs/soarca-extensions/fin-python-library.md [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+Call `set_config_MQTT_server()` to set the required configurations for the fin to connect to the MQTT broker.
+For each capability to be registered, call `create_fin_capability()`. The capability callback funtion should return an object of type `ResultStructure`.
+When all capabilities are initialized, call `start_fin()` for the SOARCA Fin to connect to the MQTT broker and register itself to SOARCA.
+
+An example is given in this project in the file `examples/pong_example.py`
```
Comment (RabbITCybErSeC) [inline docs/content/en/docs/soarca-extensions/fin-python-library.md:109]: maybe provide a clickable link here

**Review by RabbITCybErSeC — CHANGES_REQUESTED** [review]

File: docs/content/en/docs/soarca-extensions/fin-python-library.md [hunk 10]
```diff
@@ -0,0 +1,213 @@
...
+For the documentation about the Fin protocol we refer the reader to de documention page of [SOARCA Fin Protocol](https://cossas.github.io/SOARCA/docs/soarca-extensions/fin-protocol/)
+
+## Quick Start
+ 
+To include the SOARCA Fin library, you can use the following command to install it via pip:
+
+```bash
+pip install soarca-fin-library
+```
+
```
Comment (MaartendeKruijf) [inline docs/content/en/docs/soarca-extensions/fin-python-library.md:20]: Missing env definitions and that they need to match with SOARCA's configuration

**Review by MaartendeKruijf — APPROVED** [review]

**Review by RabbITCybErSeC — APPROVED** [review]

**Pull request merged**